src/test/java/seedu/contax/logic/parser/BatchCommandParserTest.java

```diff
@@ -0,0 +1,46 @@
+package seedu.contax.logic.parser;
+
+import static seedu.contax.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.contax.logic.parser.CommandParserTestUtil.assertParseFailure;
+import static seedu.contax.logic.parser.CommandParserTestUtil.assertParseSuccess;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.contax.logic.commands.BatchCommand;
+import seedu.contax.model.util.SearchType;
+
+/**
+ * Test for parses input arguments and creates a new BatchCommand object.
+ */
+public class BatchCommandParserTest {
+    private static final String MESSAGE_INVALID_FORMAT = String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+            BatchCommand.MESSAGE_USAGE);
+
+    private BatchCommandParser parser = new BatchCommandParser();
+
+    @Test
+    public void parse_missingParts_failure() {
+
+        // no index specified
+        assertParseFailure(parser, "", MESSAGE_INVALID_FORMAT);
+
+        // no regex and no field specified
+        assertParseFailure(parser, "batch edit", MESSAGE_INVALID_FORMAT);
+
+        // search type is not specified
+        assertParseFailure(parser, "batch edit =/1", MESSAGE_INVALID_FORMAT);
+
+    }
+
+    @Test
+    public void parse_someFieldsSpecified_success() {
```

Comment:
Missing test cases of all the search types.

Reply:
added